### PR TITLE
[MPLUGIN-336] Jdk11 asm7

### PR DESCRIPTION
**Pull request by olamy**

### CODE REVIEW

**Comment by olamy** [comment]
@rfscholte  @khmarbaise @hboutemy @sormuras  if you can review that (I even try to follow checkstyle :P )

File: maven-plugin-tools-generators/src/main/java/org/apache/maven/tools/plugin/generator/PluginXdocGenerator.java [hunk 1]
```diff
@@ -405,8 +395,7 @@ private void writeGoalAttributes( MojoDescriptor mojoDescriptor, XMLWriter w )
      */
     private void writeGoalParameterTable( MojoDescriptor mojoDescriptor, XMLWriter w )
     {
-        @SuppressWarnings( "unchecked" )
-        List<Parameter> parameterList = mojoDescriptor.getParameters();
+        @SuppressWarnings( "unchecked" ) List<Parameter> parameterList = mojoDescriptor.getParameters();
```
Comment (rfscholte) [inline maven-plugin-tools-generators/src/main/java/org/apache/maven/tools/plugin/generator/PluginXdocGenerator.java]: Is the suppressWarning still required?

File: maven-script/maven-plugin-tools-model/src/main/java/org/apache/maven/tools/plugin/extractor/model/PluginMetadataParser.java [hunk 2]
```diff
@@ -71,36 +67,25 @@
 
             if ( mojos != null && !mojos.isEmpty() )
```
Comment (rfscholte) [inline maven-script/maven-plugin-tools-model/src/main/java/org/apache/maven/tools/plugin/extractor/model/PluginMetadataParser.java]: `&& !mojos.isEmpty()` can be removed, let the for-loop handle this.

File: maven-plugin-plugin/pom.xml [hunk 3]
```diff
@@ -390,6 +390,25 @@
         </plugins>
       </build>
     </profile>
+    <profile>
```
[the remark this answers is not in the document]
Comment (olamy) [reply maven-plugin-plugin/pom.xml]: uhm not so sure if the argument is propagated.... see https://builds.apache.org/blue/organizations/jenkins/maven-box%2Fmaven-plugin-tools/detail/jdk11_asm7/7/pipeline/32

File: maven-plugin-tools-generators/src/main/java/org/apache/maven/tools/plugin/generator/PluginXdocGenerator.java (hunk 1 above)
Comment (olamy) [reply maven-plugin-tools-generators/src/main/java/org/apache/maven/tools/plugin/generator/PluginXdocGenerator.java]: I guess not but I didn't upgrade to maven 3 :)

**Pull request merged**